src/test/java/com/zimbra/cert/GetDomainCertTest.java

```diff
@@ -0,0 +1,136 @@
+package com.zimbra.cert;
+
+import static com.zimbra.common.soap.AdminConstants.A_DOMAIN;
+import static com.zimbra.common.soap.CertMgrConstants.GET_DOMAIN_CERT_REQUEST;
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import com.zimbra.common.account.Key.DomainBy;
+import com.zimbra.common.service.ServiceException;
+import com.zimbra.common.soap.Element;
+import com.zimbra.common.soap.Element.XMLElement;
+import com.zimbra.common.soap.SoapProtocol;
+import com.zimbra.cs.account.AuthToken;
+import com.zimbra.cs.account.Domain;
+import com.zimbra.cs.account.Provisioning;
+import com.zimbra.soap.SoapEngine;
+import com.zimbra.soap.ZimbraSoapContext;
+import java.util.HashMap;
+import java.util.Map;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+public class GetDomainCertTest {
+  private final String domainId = "domainId";
+  private final Map<String, Object> context = new HashMap<>();
+  Provisioning provisioning = mock(Provisioning.class);
+
+  @Rule public ExpectedException expectedEx = ExpectedException.none();
+
+  @Before
+  public void setUp() throws Exception {
+    final ZimbraSoapContext zsc =
+        new ZimbraSoapContext(mock(AuthToken.class), "1", SoapProtocol.Soap12, SoapProtocol.Soap12);
+    context.put(SoapEngine.ZIMBRA_CONTEXT, zsc);
+    Provisioning.setInstance(provisioning);
+  }
+
+  @Test(expected = ServiceException.class)
```

Comment:
I think checking only ServiceException is thrown is a bit generic, check also the message/code of the exception is what we want.
You can use: https://junit.org/junit4/javadoc/4.13/org/junit/rules/ExpectedException.html or https://www.baeldung.com/junit-assert-exception

Reply:
lol @frisonisland i've tried import asserThrows but there is no this method there
will check that you propose

Reply:
> lol @frisonisland i've tried import asserThrows but there is no this method there will check that you propose

Check this: https://github.com/Zextras/carbonio-mailbox/blob/main/store/src/test/java/com/zimbra/cs/service/mail/CopyToFilesIT.java

Reply:
@frisonisland yeah i've made with 
@Rule public ExpectedException expectedEx = ExpectedException.none();
 expectedEx.expect(ServiceException.class);
 expectedEx.expectMessage("Domain with id domainId could not be found.");